src/permissions.js

```diff
@@ -100,30 +73,41 @@ export const appRoles = (app, permissions) =>
     })
   )
 
-// Returns a function that resolves a role
-// using the provided apps, and caching the result.
-export const roleResolver = (apps = []) =>
-  memoize((proxyAddress, roleBytes) => {
-    const knownRole = getKnownRole(roleBytes)
-    if (knownRole) {
-      return knownRole.role
-    }
+// Resolves a role using the provided apps
+function resolveRole(apps, proxyAddress, roleBytes) {
+  const knownRole = getKnownRole(roleBytes)
+  if (knownRole) {
+    return knownRole.role
+  }
+  const app = apps.find(app => app.proxyAddress === proxyAddress)
+  if (!app || !app.roles) {
+    return null
+  }
+  return app.roles.find(role => role.bytes === roleBytes)
+}
 
-    const app = apps.find(app => app.proxyAddress === proxyAddress)
-    if (!app || !app.roles) {
-      return null
-    }
-    return app.roles.find(role => role.bytes === roleBytes)
-  }, (...args) => args[0] + args[1])
+// Resolves an entity using the provided apps
+function resolveEntity(apps, address) {
+  const entity = { address, type: 'address' }
+  if (isAnyAddress(address)) {
+    return { ...entity, type: 'any' }
+  }
+  const app = apps.find(app => app.proxyAddress === address)
+  return app ? { ...entity, type: 'app', app } : entity
+}
 
-// Returns a function that resolves an entity
-// using the provided apps, and caching the result.
-export const entityResolver = (apps = []) =>
-  memoize(address => {
-    const entity = { address, type: 'address' }
-    if (address === ANY_ADDRESS) {
-      return { ...entity, type: 'any' }
-    }
-    const app = apps.find(app => app.proxyAddress === address)
-    return app ? { ...entity, type: 'app', app } : entity
-  }, (...args) => args[0] + args[1])
+// Returns a function that resolves an entity, caching the results
+export function entityResolver(apps = []) {
+  return memoize(
+    address => resolveEntity(apps, address),
+    (...args) => args[0] + args[1]
+  )
+}
+
+// Returns a function that resolves an role, caching the results
+export function roleResolver(apps = []) {
+  return memoize(
+    (proxyAddress, roleBytes) => resolveRole(apps, proxyAddress, roleBytes),
+    (...args) => args[0] + args[1]
```

Comment:
I think for both of these memoized functions, we don't need to provide a custom resolver anymore? AFAIK we're only using them with one argument.

Reply:
This one is still having two, but you’re right for `entityResolver` 👍